From: Sergei Golubchik <serg@mariadb.org>
Date: Fri, 13 Feb 2026 15:22:57 +0100
Subject: [PATCH] MDEV-38811 crash in information_schema.table_constraints
 when --skip-grant-tables

acl_get_all3() wasn't expecting --skip-grant-tables

Multiple users reported Akonadi crashing with MariaDB 11.8.6. Downgrading to
11.8.5 fixed it. Reason most likely due to upstream regression as described
in detail in bug reports.

Origin: https://github.com/MariaDB/server/commit/87309d3d4bb8f48910d05b0ca5ee989bcdd6b053.patch
Bug: https://jira.mariadb.org/browse/MDEV-38811
Forwarded: not-needed
---
 sql/sql_acl.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sql/sql_acl.cc b/sql/sql_acl.cc
index a583c0b..b6abd27 100644
--- a/sql/sql_acl.cc
+++ b/sql/sql_acl.cc
@@ -3927,6 +3927,9 @@ privilege_t acl_get(const char *host, const char *ip,
 privilege_t acl_get_all3(Security_context *sctx, const char *db,
                          bool db_is_patern)
 {
+  if (!initialized)
+    return DB_ACLS;
+
   privilege_t access= acl_get(sctx->host, sctx->ip,
                               sctx->priv_user, db, db_is_patern);
   if (sctx->priv_role[0])
